syntax = "proto3";

package armonik.api.grpc.v1.partitions;

import "partitions_common.proto";

option csharp_namespace = "Armonik.Api.Grpc.V1.Partitions";

/**
 * The PartitionsService provides methods to manage partitions.
 */
service Partitions {
  /**
   * Get a partitions list using pagination, filters and sorting.
   */
  rpc ListPartitions(ListPartitionsRequest) returns (ListPartitionsResponse) {}

  /**
   * Get a partition by its ID.
   */
  rpc GetPartition(GetPartitionRequest) returns (GetPartitionResponse) {}

  /**
   * Update a partition by its ID.
   * Null or empty (or -1 for integers) data means that tehy will not be updated
   */
  rpc UpdatePartitions(UpdatePartitionsRequest) returns (UpdatePartitionsResponse) {}

  /**
   * Delete a list of partition by ID.
   */
  rpc DeletePartitions(DeletePartitionsRequest) returns (DeletePartitionsResponse) {}

  /**
   * Create partitions and return their IDs.
   */
  rpc CreatePartitions(CreatePartitionsRequest) returns (CreatePartitionsResponse) {}
}
